src/new/wallet.js

```diff
@@ -2322,6 +2323,53 @@ class HathorWallet extends EventEmitter {
     const tx = await this.prepareCreateNewToken(name, symbol, amount, newOptions);
     return this.handleSendPreparedTransaction(tx);
   }
+
+  /**
+   * Sign all inputs
+   *
+   * @param {Transaction} tx The transaction to be signed
+   * @param [options]
+   * @param {string} [options.pincode] PIN to decrypt the private key.
+   *                                   Optional but required if not set in this
+   *
+   * @returns {Transaction} The signed transaction
+   */
+  getSignatures(tx, { pinCode = null } = {}) {
+    if (this.isFromXPub()) {
+      throw new WalletFromXPubGuard('getSignatures');
+    }
+    storage.setStore(this.store);
+    const pin = pinCode || this.pinCode;
+    if (!pin) {
+      throw new Error(ERROR_MESSAGE_PIN_REQUIRED);
+    }
+
+    // get private key
+    const accessData = storage.getItem('wallet:accessData');
+    const privateKeyStr = wallet.decryptData(accessData.mainKey, pin);
+    const key = HDPrivateKey(privateKeyStr);
+
+    for (const input of tx.inputs) {
+      const inputTx = this.getTx(input.hash);
+      if (!inputTx) {
+        // Input is not from our wallet
+        continue;
+      }
+      const utxo = inputTx.outputs[input.index];
+      if(utxo && this.isAddressMine(utxo.decoded.address)) {
```

Comment:
We should have a linter for this later, it's just for consistency

```suggestion
      if (utxo && this.isAddressMine(utxo.decoded.address)) {
```

Reply:
Done